corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js

```diff
@@ -197,6 +197,31 @@ FreeTextEntry.prototype.onPreProcess = function (newValue) {
     this.question.error(this.getErrorMessage(newValue));
 };
 
+function AddressEntry(question, options) {
+    var self = this;
+    FreeTextEntry.call(this, question, options);
+    this.templateType = 'address';
+    this.lengthLimit = options.lengthLimit;
+
+    self.afterRender = function () {
+        var accessToken = 'pk.eyJ1Ijoia25ndXllbi1kaW1hZ2kiLCJhIjoiY2tjM3ZtbXltMDF3OTJ5bnd0cjBpeHc4NSJ9.fVcAJi5sxr-vi_UQf2RnkA';
+        var geocoder = new MapboxGeocoder({
+            accessToken: accessToken,
+            types: 'address',
+            proximity: { longitude: -74.006058, latitude: 40.712772},
```

Comment:
My read of the docs is that this means locations nearer to NYC will be weighted higher, is this correct? NYC seems good for the project requesting this, maybe less good for other US projects, and not good for international projects. Do you know how much the proximity value affects the search results? If it's a lot, we might want to consider alternatives (send location based on user's IP, maybe?).

Reply:
Yes, I forgot to make a comment on this.  Yeah I didn't know which one proximity to choose initially and I was getting far away results without it. I think it's similar to the geowidget where we had to select an initial location to begin with. With it set to NYC, I was still able to get international results if you're specific. You can play around with it [here](https://jsfiddle.net/28v51xuj/) 

I'm not familiar with how locations are usually requested from the browser, but there's usually a `request location` permission that goes to the server, right? Since this is only run on the browser, do we just make an ajax request to an ip service to get the ip?

Reply:
That was actual just by their request. But maybe we should strip out all `<country>-<region>`?

Reply:
We are going to use this widget pretty broadly across the US, so I think we'll be sad if we hard code to NY without a way to override..

I think the easiest default would be to make this a project space wide setting, although I don't know how easy it is for the page to inject RequireJS context into this part of formplayer (@orangejenny do you know that?). I found it pretty difficult to get context info fed down into the engine.

I have a recent example of adding settings models in here that could be reviewed
https://github.com/dimagi/commcare-hq/pull/28089/files#diff-de4965efa3b30eb36bffc0cbd0f8e0b9

(and for clarity, I don't think it's a blocker to accept the PR with NY hard coded, just taht we should know what our next move will be for a broader solution.

Reply:
I was thinking of an ajax request to a location service, yes, but a project space setting sounds fine if that's easier to implement. Also agree this isn't blocking feedback.

A project space setting could be made available to javascript via initial page data. It should be fine in this part of formplayer; `reverse`, which is just a specific type of initial page data, is used [here](https://github.com/dimagi/commcare-hq/blob/afe39c4bcb3c811b7a1543269bf69c47a69869c4/corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js#L618).